src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -334,6 +342,21 @@ class IpexCommunicationService extends AgentService {
       },
     });
   }
+
+  async createLinkedIpexMessageRecord(
```

Comment:
Can we update this to accept the `schemaSaid` and fetch the schema here? Would be a bit tidier.

Also, for schemas, right now:
- We resolve them off the test credential server by schemaSaid
- Once available, we can call `schemas().get()`

In this function, can you handle the case where `schemas().get()` returns a 404, and try to resolve the schema using the URL (see `admitAcdc` where we resolve schemas). If that fails, it's still worth storing it.

Perhaps, we can also store the `schema` in the IPEX record. So if `credentialType` is undefined, we can say it's unknown right now - and we can later update it using the schema SAID. (that would be another story)